DAQ_System/Source/main.cpp

```diff
@@ -46,154 +51,63 @@ int main() {
+        snprintf(buffer, sizeof(buffer), "%f, %d, %d, %d, %d\n", ((float)timer)/CLOCKS_PER_SEC, linpot1, linpot2, linpot3, linpot4);
+        FileWrite(data_file, buffer);
+        linpot1++;
+        linpot2++;
+        linpot3++;
+        linpot4++;
+        timer = clock() - timer;
     }
 
-    printf("\rIncrementing numbers (%d/%d)... OK\n", 10, 10);
-
     // Close the file which also flushes any cached writes
-    printf("Closing \"/fs/numbers.txt\"... ");
+    printf("Closing \"/fs/data.csv\"... ");
     fflush(stdout);
-    FileClose(numbers_file);
-
-
-    // Display the root directory
-    printf("Opening the root directory... ");
-    fflush(stdout);
-
-    DIR* root_dir = opendir("/fs/");
-    printf("%s\n", (!root_dir ? "Fail :(" : "OK"));
-    if (!root_dir) {
-        error("error: %s (%d)\n", strerror(errno), -errno);
-    }
-
-    printf("root directory:\n");
-    while (true) {
-        struct dirent* entry = readdir(root_dir);
-        if (!entry) {
-            break;
-        }
-
-        printf("    %s\n", entry->d_name);
-    }
-
-    printf("Closing the root directory... ");
-    fflush(stdout);
-    status = closedir(root_dir);
-    printf("%s\n", (status < 0 ? "Fail :(" : "OK"));
-    if (status < 0) {
-        error("error: %s (%d)\n", strerror(errno), -errno);
-    }
-
-    // Display the numbers file
-    printf("Opening \"/fs/numbers.txt\"... ");
-    fflush(stdout);
-    numbers_file = FileOpen("/fs/numbers.txt");
-
-    printf("numbers:\n");
-    while (!feof(numbers_file)) {
-        int num = fgetc(numbers_file);
-        printf("%c", num);
-    }
-
-    printf("\rClosing \"/fs/numbers.txt\"... ");
-    fflush(stdout);
-    FileClose(numbers_file);
+    FileClose(data_file);
 
     // Tidy up
     printf("Unmounting... ");
     fflush(stdout);
     Unmount(&file_system);
     
-    printf("Initializing the block device... ");
-    fflush(stdout);
-
-    status = block_device.init();
-    printf("%s\n", (status ? "Fail :(" : "OK"));
-    if (status) {
-        error("error: %s (%d)\n", strerror(-status), status);
-    }
-
-    printf("Erasing the block device... ");
-    fflush(stdout);
-    status = block_device.erase(0, block_device.size());
-    printf("%s\n", (status ? "Fail :(" : "OK"));
-    if (status) {
-        error("error: %s (%d)\n", strerror(-status), status);
-    }
-
-    printf("Deinitializing the block device... ");
-    fflush(stdout);
-    status = block_device.deinit();
-    printf("%s\n", (status ? "Fail :(" : "OK"));
-    if (status) {
-        error("error: %s (%d)\n", strerror(-status), status);
-    }
-    
     printf("\r\n");
     
-    printf("Mbed OS filesystem example done!\n");
+    printf("CSV File Test Done!\n");
 }
 
 // mounts the filesystem to the given sd block device and checks for errors, reformats the device of no filesystem is found
-void Mount(FATFileSystem *file_system, SDBlockDevice *block_device) {
+int Mount(FATFileSystem *file_system, SDBlockDevice *block_device) {
```

Comment:
Change the type of `status` and the function return type to `uint8_t` or `bool`.  Do this for the other functions, too.